homeassistant/components/zwave_js/helpers.py

```diff
@@ -395,12 +395,15 @@ def async_is_device_config_entry_not_loaded(
     dev_reg = dr.async_get(hass)
     if (device := dev_reg.async_get(device_id)) is None:
         raise ValueError(f"Device {device_id} not found")
-    return any(
-        (entry := hass.config_entries.async_get_entry(entry_id))
-        and entry.domain == DOMAIN
-        and entry.state != ConfigEntryState.LOADED
-        for entry_id in device.config_entries
-    )
+    try:
+        entry = next(
+            config_entry
+            for config_entry in hass.config_entries.async_entries(DOMAIN)
+            if config_entry.entry_id in device.config_entries
+        )
+    except StopIteration as err:
+        raise ValueError(f"Device {device_id} has no Z-Wave JS config entry") from err
```

Comment:
How will this play with config validation before Home Assistant is running? We want to handle that case gracefully and just return the config without trying further validation.

Existing code:
https://github.com/home-assistant/core/blob/554f079b0244a582e4244ec6ad7654c617ffbcb3/homeassistant/components/zwave_js/device_trigger.py#L215-L227

Maybe raise another type of exception for that case and handle that appropriately?

Reply:
Would the config entry not be found at start up? I assumed it would be there, just not loaded

Reply:
Not if Home Assistant isn't running at all.

Reply:
hmm ok

Reply:
OK so I looked at the code, and the config entries always get loaded before the device registry unless I am missing something.

1. HA setup function: https://github.com/home-assistant/core/blob/dev/homeassistant/bootstrap.py#L93
2. Start configuring HA from config: https://github.com/home-assistant/core/blob/dev/homeassistant/bootstrap.py#L138
3. Load config entries: https://github.com/home-assistant/core/blob/554f079b0244a582e4244ec6ad7654c617ffbcb3/homeassistant/bootstrap.py#L218-L219
4. Start setting up integrations: https://github.com/home-assistant/core/blob/554f079b0244a582e4244ec6ad7654c617ffbcb3/homeassistant/bootstrap.py#L251
5. Load the registries: https://github.com/home-assistant/core/blob/554f079b0244a582e4244ec6ad7654c617ffbcb3/homeassistant/bootstrap.py#L545
6. Setup integrations, which would include automations: https://github.com/home-assistant/core/blob/554f079b0244a582e4244ec6ad7654c617ffbcb3/homeassistant/bootstrap.py#L550-L572

Based on that order of operations, if this was a problem, wouldn't the lack of a device in the device registry be causing a problem? If there's something I am missing, just tell me that and I'll move on, sorry to drag this out